myriad-scheduler/src/main/java/org/apache/myriad/webapp/HttpConnectorProvider.java

```diff
@@ -38,11 +53,55 @@ public HttpConnectorProvider(MyriadConfiguration myriadConf) {
 
   @Override
   public Connector get() {
-    SelectChannelConnector ret = new SelectChannelConnector();
+    final AbstractConnector ret;
+    if (!myriadConf.isSSLEnabled()) {
+      final SelectChannelConnector listener = new SelectChannelConnector();
+      ret = listener;
+    } else {
+      final Configuration sslConf = new Configuration(false);
+      boolean needsClientAuth = YarnConfiguration.YARN_SSL_CLIENT_HTTPS_NEED_AUTH_DEFAULT;
+      sslConf.addResource(YarnConfiguration.YARN_SSL_SERVER_RESOURCE_DEFAULT);
+
+      final SslSocketConnectorSecure listener = new SslSocketConnectorSecure();
+      listener.setHeaderBufferSize(1024 * 64);
+      listener.setNeedClientAuth(needsClientAuth);
+      listener.setKeyPassword(getPassword(sslConf, WEB_APP_KEY_PASSWORD_KEY));
+
+      String keyStore = sslConf.get("ssl.server.keystore.location");
+
+      if (keyStore != null) {
+        listener.setKeystore(keyStore);
+        listener.setKeystoreType(sslConf.get("ssl.server.keystore.type", "jks"));
+        listener.setPassword(getPassword(sslConf, WEB_APP_KEYSTORE_PASSWORD_KEY));
+      }
+
+      String trustStore = sslConf.get("ssl.server.truststore.location");
+
+      if (trustStore != null) {
+        listener.setTruststore(trustStore);
+        listener.setTruststoreType(sslConf.get("ssl.server.truststore.type", "jks"));
+        listener.setTrustPassword(getPassword(sslConf, WEB_APP_TRUSTSTORE_PASSWORD_KEY));
+      }
+      ret = listener;
+    }
+
     ret.setName("Myriad");
     ret.setHost("0.0.0.0");
     ret.setPort(myriadConf.getRestApiPort());
-
     return ret;
   }
+
+  static String getPassword(Configuration conf, String alias) {
+    String password = null;
+    try {
+      char[] passchars = conf.getPassword(alias);
+      if (passchars != null) {
+        password = new String(passchars);
+      }
+    } catch (IOException ioe) {
+      password = null;
```

Comment:
Those params are hadoop Configuration params, not Myriad ones and I don't think we should duplicate them in Myriad configuration.